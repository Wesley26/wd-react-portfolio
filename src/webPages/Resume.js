import React from 'react';
import { noImage, 
        resumeImg,
        resumeLink,
        resumeText } from './contentMaster/PageContents.js';

const Resume = () => {
    
    return(
        <div className="font-body bg-body-lightGray m-6 sm:pb-6 shadow-xl">

            <div className="px-5 py-3">
                <div className="text-xl p-3 text-center">
                    <p>
                        {resumeText.infoR}
                    </p>
                </div>
                <img 
                    src={resumeImg} 
                    alt={noImage}
                    className="focus:outline-none shadow-lg">
                </img>
            </div>
            <div className="pt-3 pb-12 md:ml-64 md:mr-64 lg:ml-64 lg:mr-64 xl:ml-32long xl:mr-32long grid grid-flow-col">
                <div className="text-xl p-3 flex rounded-full justify-center hover:bg-gray-300 active:bg-gray-500">
                    <a href={resumeLink}>
                        {resumeText.downloadR}
                    </a>
                </div>
            </div>

            <div className="sm:visible sm:mb-32long md:hidden">
                <br></br>
                <br></br>
                <br></br>
                <br></br>
            </div>

        </div>
    )
};

export default Resume;