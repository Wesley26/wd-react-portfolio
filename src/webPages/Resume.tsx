import { FC, useEffect, useState } from 'react';
import PageContents from './contentMaster/PageContents';
import MainPageAnimations from './contentMaster/MainPageAnimations';
import { isMobile } from 'react-device-detect';
import { motion } from 'framer-motion';

const Resume:FC = () => {

    const pageContents = PageContents();
    const mainPageAnimations = MainPageAnimations();

    const [visible, setVisible] = useState<boolean>(false);

    useEffect(() => {

        setVisible(false);
        setVisible(true);

    }, []);
    
    return(
        <div className="font-body bg-body_lightGray m-6 sm:pb-6 shadow-xl h-full">

            <> { visible ? (
                <motion.div
                    initial={mainPageAnimations.initialStyle}
                    animate={mainPageAnimations.animateStyle}
                    transition={mainPageAnimations.transitionStyle}
                    className="px-5 py-3"
                >
                    <div className="text-xl p-3 text-center">
                        <p>
                            {pageContents.resumeText.infoR}
                        </p>
                    </div>
                    <div className="flex flex-1 justify-center items-center h-30long md:h-51.6long lg:h-85.6long">
                        { isMobile ? ( //Resume
                            <img 
                                src={pageContents.allImages.resumePdf}
                                alt={pageContents.noImage}
                                className="focus:outline-none shadow-lg"
                            /> 
                        ) : (
                            <iframe 
                                src={`${pageContents.allImages.resumePdf}#view=fitH#zoom=scale`}
                                title={pageContents.resumeText.resumeTitle}
                                className="focus:outline-none shadow-lg h-full w-full"
                                height="100%"
                                width="100%"
                            />
                        )}
                    </div>
                    <>
                        <div className="pt-3 pb-12 lg:ml-64 lg:mr-64 grid grid-flow-col">
                            <div className="text-xl p-3 flex rounded-full justify-center hover:bg-gray-300 active:bg-gray-500">
                                <a href={pageContents.resumeLink} rel="noreferrer" target="_blank">
                                    {pageContents.resumeText.downloadR}
                                </a>
                            </div>
                        </div>
                        <div className="sm:visible sm:mb-32long md:hidden">
                            <br></br>
                        </div>
                    </>
                    <div className="flex flex-1 justify-center items-center h-30long md:h-51.6long lg:h-85.6long">
                        { isMobile ? ( //ICM Letter of Recommendation
                            <img 
                                src={pageContents.allImages.letterBImg}
                                alt={pageContents.noImage}
                                className="focus:outline-none shadow-lg h-93percent"
                            /> 
                        ) : (
                            <iframe 
                                src={`${pageContents.allImages.letterBPdf}#view=fitH#zoom=scale`}
                                title={pageContents.resumeText.letterOfRecommendationTitleB}
                                className="focus:outline-none shadow-lg h-full w-full"
                                height="100%"
                                width="100%"
                            />
                        )}
                    </div>
                    <div className="flex flex-1 justify-center items-center h-30long md:h-51.6long lg:h-85.6long">
                        { isMobile ? ( //CHOYCES Letter of Recommendation
                            <img 
                                src={pageContents.allImages.letterAImg}
                                alt={pageContents.noImage}
                                className="focus:outline-none shadow-lg h-93percent"
                            /> 
                        ) : (
                            <iframe 
                                src={`${pageContents.allImages.letterAPdf}#view=fitH#zoom=scale`}
                                title={pageContents.resumeText.letterOfRecommendationTitleA}
                                className="focus:outline-none shadow-lg h-full w-full"
                                height="100%"
                                width="100%"
                            />
                        )}
                    </div>
                    <>
                        <div className="sm:visible sm:mb-32long md:hidden">
                            <br></br>
                            <br></br>
                            <br></br>
                            <br></br>
                        </div>
                    </>
                    </motion.div>
                ) : null }
            </>

        </div>
    )
};

export default Resume;