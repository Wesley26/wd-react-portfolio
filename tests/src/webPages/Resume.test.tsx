import { render } from '@testing-library/react';
import Resume from '../../../src/webPages/Resume';
import '../../src/index.css';

it(`render ContactMe.tsx test component`, () => {

    const view = (render(<Resume />));

    expect(view).toMatchSnapshot();

});