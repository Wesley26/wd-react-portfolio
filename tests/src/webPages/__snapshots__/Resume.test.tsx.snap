// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render ContactMe.tsx test component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="font-body bg-body_lightGray m-6 sm:pb-6 shadow-xl h-full"
      >
         
        <div
          class="px-5 py-3"
          style="opacity: 0; transform: translateX(0px) translateY(-20px) translateZ(0);"
        >
          <div
            class="text-xl p-3 text-center"
          >
            <p>
              To view the full PDF file of my resume and supporting documents, select the download link at the bottom of this page.
            </p>
          </div>
          <div
            class="flex flex-1 justify-center items-center h-30long md:h-51.6long lg:h-85.6long"
          >
            <iframe
              class="focus:outline-none shadow-lg h-full w-full"
              height="100%"
              src="[object Object]#view=fitH#zoom=scale"
              title="WD_Resume"
              width="100%"
            />
          </div>
          <div
            class="pt-3 pb-12 lg:ml-64 lg:mr-64 grid grid-flow-col"
          >
            <div
              class="text-xl p-3 flex rounded-full justify-center hover:bg-gray-300 active:bg-gray-500"
            >
              <a
                href="https://drive.google.com/drive/folders/1Zq6rcMgDz2IeEZCGJr4lKEqATsdVakbW?usp=sharing"
                rel="noreferrer"
                target="_blank"
              >
                Download My Resume and Supporting Documents
              </a>
            </div>
          </div>
          <div
            class="sm:visible sm:mb-32long md:hidden"
          >
            <br />
          </div>
          <div
            class="flex flex-1 justify-center items-center h-30long md:h-51.6long lg:h-85.6long"
          >
            <iframe
              class="focus:outline-none shadow-lg h-full w-full"
              height="100%"
              src="[object Object]#view=fitH#zoom=scale"
              title="WD_Letter_Of_Recommendation_B"
              width="100%"
            />
          </div>
          <div
            class="flex flex-1 justify-center items-center h-30long md:h-51.6long lg:h-85.6long"
          >
            <iframe
              class="focus:outline-none shadow-lg h-full w-full"
              height="100%"
              src="[object Object]#view=fitH#zoom=scale"
              title="WD_Letter_Of_Recommendation_A"
              width="100%"
            />
          </div>
          <div
            class="sm:visible sm:mb-32long md:hidden"
          >
            <br />
            <br />
            <br />
            <br />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="font-body bg-body_lightGray m-6 sm:pb-6 shadow-xl h-full"
    >
       
      <div
        class="px-5 py-3"
        style="opacity: 0; transform: translateX(0px) translateY(-20px) translateZ(0);"
      >
        <div
          class="text-xl p-3 text-center"
        >
          <p>
            To view the full PDF file of my resume and supporting documents, select the download link at the bottom of this page.
          </p>
        </div>
        <div
          class="flex flex-1 justify-center items-center h-30long md:h-51.6long lg:h-85.6long"
        >
          <iframe
            class="focus:outline-none shadow-lg h-full w-full"
            height="100%"
            src="[object Object]#view=fitH#zoom=scale"
            title="WD_Resume"
            width="100%"
          />
        </div>
        <div
          class="pt-3 pb-12 lg:ml-64 lg:mr-64 grid grid-flow-col"
        >
          <div
            class="text-xl p-3 flex rounded-full justify-center hover:bg-gray-300 active:bg-gray-500"
          >
            <a
              href="https://drive.google.com/drive/folders/1Zq6rcMgDz2IeEZCGJr4lKEqATsdVakbW?usp=sharing"
              rel="noreferrer"
              target="_blank"
            >
              Download My Resume and Supporting Documents
            </a>
          </div>
        </div>
        <div
          class="sm:visible sm:mb-32long md:hidden"
        >
          <br />
        </div>
        <div
          class="flex flex-1 justify-center items-center h-30long md:h-51.6long lg:h-85.6long"
        >
          <iframe
            class="focus:outline-none shadow-lg h-full w-full"
            height="100%"
            src="[object Object]#view=fitH#zoom=scale"
            title="WD_Letter_Of_Recommendation_B"
            width="100%"
          />
        </div>
        <div
          class="flex flex-1 justify-center items-center h-30long md:h-51.6long lg:h-85.6long"
        >
          <iframe
            class="focus:outline-none shadow-lg h-full w-full"
            height="100%"
            src="[object Object]#view=fitH#zoom=scale"
            title="WD_Letter_Of_Recommendation_A"
            width="100%"
          />
        </div>
        <div
          class="sm:visible sm:mb-32long md:hidden"
        >
          <br />
          <br />
          <br />
          <br />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
